src/MongoDB/BSON/ObjectID.cpp

```diff
@@ -44,19 +58,8 @@ void HHVM_METHOD(MongoDBBsonObjectID, __construct, const Variant &objectId)
 	} else {
 		bson_oid_init(&data->m_oid, NULL);
 	}
-}
-
-static String oidAsString(MongoDBBsonObjectIDData* data)
-{
-	String s;
-	char *data_s;
 
-	s = String(24, ReserveString);
-	data_s = s.bufferSlice().data();
-	bson_oid_to_string(&data->m_oid, data_s);
-	s.setSize(24);
-
-	return s;
+	this_->o_set(s_MongoBsonObjectID_oid, Variant(oidAsString(data)), s_MongoBsonObjectID_className);
```

Comment:
Are you storing the hexadecimal string as a property to keep the userland methods from relying on any C++ methods? If `oidAsString()` was repurposed to the implementation for `__toString()`, `__debugInfo()` and `serialize()` could just string-cast `$this` to obtain the value without requiring a separate property.